JustSaying/AwsTools/QueueCreation/RegionResourceCache.cs

```diff
@@ -1,33 +1,36 @@
 using System.Collections.Generic;
 
 namespace JustSaying.AwsTools.QueueCreation
-{
-    public class RegionResourceCache<T> : Dictionary<string, Dictionary<string, T>>, IRegionResourceCache<T>
+{
+    public sealed class RegionResourceCache<T>
     {
+        private readonly Dictionary<string, Dictionary<string, T>> _regionsData
+            = new Dictionary<string, Dictionary<string, T>>();
+
         public T TryGetFromCache(string region, string key)
         {
-            if (! ContainsKey(region))
+            if (!_regionsData.ContainsKey(region))
```

Comment:
Use `TryGetValue()`. It's cheaper than a `Contains()` call and then a `[]` call,  and is race-condition safe.